apps/webapp/app/services/deleteProject.server.ts

```diff
@@ -0,0 +1,142 @@
+import { PrismaClient } from "@trigger.dev/database";
+import { prisma } from "~/db.server";
+import { DisableJobService } from "./jobs/disableJob.server";
+import { AuthenticatedEnvironment } from "./apiAuth.server";
+import { DeleteJobService } from "./jobs/deleteJob.server";
+import { DeleteEndpointService } from "./endpoints/deleteEndpointService";
+import { logger } from "./logger.server";
+import { DisableScheduleSourceService } from "./schedules/disableScheduleSource.server";
+
+type Options = { projectId: string; userId: string } | { projectSlug: string; userId: string };
+
+export class DeleteProjectService {
+  #prismaClient: PrismaClient;
+
+  constructor(prismaClient: PrismaClient = prisma) {
+    this.#prismaClient = prismaClient;
+  }
+
+  public async call(options: Options) {
+    const projectId = await this.#getProjectId(options);
+    const project = await this.#prismaClient.project.findFirst({
+      include: {
+        environments: {
+          include: {
+            endpoints: true,
+          },
+        },
+        jobs: {
+          where: { deletedAt: null },
+          include: {
+            aliases: {
+              where: {
+                name: "latest",
+              },
+              include: {
+                version: true,
+              },
+              take: 1,
+            },
+          },
+        },
+        organization: true,
+      },
+      where: {
+        id: projectId,
+        organization: { members: { some: { userId: options.userId } } },
+      },
+    });
+
+    if (!project) {
+      throw new Error("Project not found");
+    }
+
+    if (project.deletedAt) {
+      throw new Error("Project already deleted");
+    }
+
+    //disable and delete all jobs
+    const service = new DisableScheduleSourceService();
+    for (const environment of project.environments) {
+      //disable the event dispatchers
+      await this.#prismaClient.eventDispatcher.updateMany({
+        where: {
+          environmentId: environment.id,
+        },
+        data: {
+          enabled: false,
+        },
+      });
+      const eventDispatchers = await this.#prismaClient.eventDispatcher.findMany({
+        where: {
+          environmentId: environment.id,
+        },
+      });
+
+      logger.info("Deleting jobs", { jobs: project.jobs });
+      for (const job of project.jobs) {
+        //disable all the job versions
+        await this.#prismaClient.jobVersion.updateMany({
+          where: {
+            jobId: job.id,
+          },
+          data: {
+            status: "DISABLED",
+          },
+        });
+
+        await this.#prismaClient.job.update({
+          where: {
+            id: job.id,
+          },
+          data: {
+            deletedAt: new Date(),
+          },
+        });
+
+        //disable scheduled sources
+        for (const eventDispatcher of eventDispatchers) {
+          await service.call({
+            key: job.id,
```

Comment:
Are we sure this key value is correct here?

Reply:
It's taken from the disabling job code here: https://github.com/triggerdotdev/trigger.dev/blob/a9ff32418e4be194a76a91a195c782f7ae98e28f/apps/webapp/app/services/jobs/disableJob.server.ts#L99